### PR TITLE
Changelog for Clippy 1.79 🎓

### PR DESCRIPTION
**Pull request by xFrednet**
Two cat ears from waffle,
A tail and a dress,
That's our Jyn,
The magnificent cat

~ =^.^=

---

### The cat of this release is: *Jyn* submitted by @jyn514:

The cat(s) of this Clippy release

Cats for the next release can be nominated in the comments :D

---

changelog: none

### CODE REVIEW

**Comment by jyn514** [comment]
:3

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,11 +6,73 @@ document.
...
+* Moved [`manual_clamp`] to `complexity` (From `nursery` now warn-by-default)
+  [#12543](https://github.com/rust-lang/rust-clippy/pull/12543)
+* Moved [`mixed_attributes_style`] to `style` (From `suspicious`)
+  [#12572](https://github.com/rust-lang/rust-clippy/pull/12572)
+* Moved [`readonly_write_lock`] to `perf` (From `nursery` now warn-by-default)
+  [#12479](https://github.com/rust-lang/rust-clippy/pull/12479)
+
+### Enhancements
+
+* [`module_name_repetition`]: Added the [`allowed-prefixes`] configuration to allow common prefixes.
```
[the remark this answers is not in the document]
Comment (dswij) [reply CHANGELOG.md:47]: [CI](https://github.com/rust-lang/rust-clippy/actions/runs/9383691180/job/25837888341#step:6:42) is failing on this, fyi.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,11 +6,73 @@ document.
...
+* Added [`const_is_empty`] to `suspicious`
+  [#12310](https://github.com/rust-lang/rust-clippy/pull/12310)
+
+### Moves and Deprecations
+
+* Moved [`box_default`] to `style` (From `perf`)
+  [#12601](https://github.com/rust-lang/rust-clippy/pull/12601)
+* Moved [`manual_clamp`] to `complexity` (From `nursery` now warn-by-default)
+  [#12543](https://github.com/rust-lang/rust-clippy/pull/12543)
+* Moved [`mixed_attributes_style`] to `style` (From `suspicious`)
```
Comment (y21) [inline CHANGELOG.md:40]: This change was beta-nominated and already made it into the last release ([link](https://github.com/rust-lang/rust-clippy/pull/12735#discussion_r1583668654))

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,11 +6,73 @@ document.
...
+### False Positive Fixes
+
+* [`cast_lossless`]: No longer lints when casting to `u128`
+  [#12496](https://github.com/rust-lang/rust-clippy/pull/12496)
+* [`std_instead_of_core`] No longer lints on modules that are only in `std`
+  [#12447](https://github.com/rust-lang/rust-clippy/pull/12447)
+
+### ICE Fixes
+
+* [`ptr_as_ptr`]: No longer crashes when the result of a function is casted
```
Comment (y21) [inline CHANGELOG.md:68]: Same with this one

**Review by dswij — APPROVED** [review]
Thanks for putting together the changelog!

**Comment by pauliesnug** [comment]
> :3

:3

**Pull request merged**